src/versionbits.cpp

```diff
@@ -74,6 +78,8 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
                 }
                 if (count >= nThreshold) {
                     stateNext = ThresholdState::LOCKED_IN;
+                } else if (height >= height_timeout) {
+                    stateNext = ThresholdState::FAILED;
                 }
                 break;
```

Comment:
Testing for lockin before failed seems like a semantic change?  

Moving the "height >= height_timeout" to the end means we actually check for lockin at the timeout_height, which is one block too many?

Reply:
This is the intended behaviour of BIP8, and necessary for MUST_SIGNAL to work correctly.